.buildkite/scripts/lifecycle/pre_command.sh

```diff
@@ -151,6 +151,11 @@ export GCS_SA_CDN_QA_EMAIL
 GCS_SA_CDN_QA_BUCKET="$(vault_get gcs-sa-cdn-qa bucket)"
 export GCS_SA_CDN_QA_BUCKET
 
+GOOGLE_APPLICATION_CREDENTIALS="$(mktemp -d)/kibana-gcloud-service-account.json"
```

Comment:
In https://github.com/elastic/kibana/pull/173159 @jbudz set up a similar scheme of access to a bucket for CDN assets (just a few lines above). 

What should our approach be here? Should we have specific service accounts for different usage sites, with well-controlled bucket access, and assume the specific ones right before the uploads? Or should we use a powerful service account for most of the GCS access throughout our build jobs?

Reply:
IMHO we should go with the former one. Besides the nicer access control , its more explicit and a better operational/security practise

Reply:
While I agree with the general sentiment, that granular access control is nicer, but wouldn't this get over-board with all these sites requiring 1-1 extra service account, with their own access to an individual bucket? 
```
 coverage (.buildkite/scripts/steps/code_coverage/reporting/uploadPrevSha.sh)
 SO object migration (.buildkite/scripts/steps/archive_so_migration_snapshot.sh)
 Upload static site (.buildkite/scripts/steps/code_coverage/reporting/uploadStaticSite.sh)
 ES Snapshot manifest upload (.buildkite/scripts/steps/es_snapshots/create_manifest.ts)
 Scalability? (.buildkite/scripts/steps/functional/scalability_dataset_extraction.sh)
 Benchmarking (.buildkite/scripts/steps/scalability/benchmarking.sh)
 Webpack bundle analyzer (.buildkite/scripts/steps/webpack_bundle_analyzer/upload.ts)
 Build chromium (x-pack/build_chromium/build.py)
+ bazel cache
 ```

Reply:
Sharing in case there's any interest:

https://cloud.google.com/iam/docs/best-practices-service-accounts#choose-when-to-use suggests a combination of both.  Default application credentials with permission to impersonate service accounts for specific areas.

https://cloud.google.com/docs/authentication/use-service-account-impersonation

Reply:
I like this idea, but it's not far from managing and assuming roles manually. We'll still have to manage (probably through terraform as you suggested) the sevice accounts - we'd save on generating/storing the keys though.
I'll see how difficult is it to get the default account set up with rights to impersonate.

Reply:
The `gcloud` service account we get by default in the new agents is this one: [vault-gcp-authentication@elastic-ci-prod.iam.gserviceaccount.com
](https://console.cloud.google.com/iam-admin/serviceaccounts/details/vault-gcp-authentication@elastic-ci-prod.iam.gserviceaccount.com?project=elastic-ci-prod&supportedpurview=project) - this probably shouldn't have all the rights to impersonate any other service accounts, it looks like it's limited to accessing vault. 

So if we'd like to have a power-user who can impersonate the bucket-limited service accounts, we'd need +1 account. Hopefully, I could set it up so that the account is activated in the pre-command hooks, so at the call sites, one would just need to use an impersonation flag, or command - then we wouldn't need to store/retrieve all the individual keys for the service accounts, just know their names.

@jbudz / @mistic - does this plan look OK? If so, I'll go ahead and create, setup the new account with rights.

Reply:
Agree with the plan, sounds good.

Reply:
@jbudz - should I also move the CDN account's use to this form of impersonation, and include it to the set of accounts managed together by terraform?

Reply:
(We discussed on slack, we're holding off for now)